src/Components/App/App.js

```diff
@@ -21,48 +23,63 @@ class App extends Component {
     getAllMovies()     
       .then(movies => {
         console.log('Movies Request Successful', movies)
-        this.setState({ movies: movies, isLoading: false })
+        this.setState({ movies: movies.movies, isLoading: false })
       })
       .catch(error => {
         console.log('Movies Request Failed', error)
-        this.setState({ error: "Oops! Something went wrong!" })
+        this.setState({ error: 'Oops! Something went wrong!' })
       })
   }
 
-  handleClick = (id) => {
-    const current = this.state.movies.movies.find(movie => movie.id === id)
-    getSingleMovie(current.id)
-      .then(movie => this.setState({currentMovie: movie}))
-  }
+  // componentDidUpdate = () => {
+  //   if (this.state.currentMovie && this.state.isLoading) {
+  //     this.setState({ isLoading: false })
+  //   }
+  // }
 
-  exitDetails = () => {
-    this.setState({currentMovie: ''})
+  getSingleMovieData = (id) => {
+    this.setState({ isLoading: true })
+    const selectedMovieDetails = getSingleMovie(id)
+    const selectedMovieVideos = getSingleMovieVideo(id)
+    return Promise.all([selectedMovieDetails, selectedMovieVideos])
+      .then(movie => {
+        this.setState({ currentMovie: [movie[0], movie[1]], isLoading: false })
+      })
+      .then(() => console.log('fetch:',this.state.currentMovie))
   }
 
   render() {
     return (
       <div className='App'>
         <Header />
-
-        {this.state.isLoading && !this.state.error &&
-        ( <h2 className="userMsg">Loading...</h2> )}
+        <Switch>
+        {/* {this.state.isLoading && !this.state.error &&
+        ( <h2 className="userMsg">Loading...</h2> )} */}
```

Comment:
I'm prettyyyy sure we won't end up needing this.

Reply:
same